### PR TITLE
DONT_MERGE - Multiple fsharp core dotnetcore references

### PR DESCRIPTION
**Pull request by matthid**
Trying to reproduce https://github.com/fsprojects/Paket/issues/1815

@forki I stand corrected. Paket does in fact the correct thing already... However I used the public API wrong. Now the question is how can I do the same via the public Paket API?

See the test code for details (will add some inline-comments)

### CODE REVIEW

File: integrationtests/Paket.IntegrationTests/InstallSpecs.fs [hunk 1]
```diff
@@ -508,4 +508,43 @@ let ``#1779 net20 only in net461``() =
...
+        package)
+      |> Seq.toList
+      |> Paket.LoadingScripts.PackageAndAssemblyResolution.getPackageOrderResolvedPackage
+      |> Seq.collect (fun p ->
+        let installModel = paketDependencies.GetInstalledPackageModel(group, p.Name.ToString())
+        Paket.LoadingScripts.PackageAndAssemblyResolution.getDllsWithinPackage framework installModel)
+      |> Seq.map (fun fi -> fi.FullName)
+      |> Seq.filter (fun fi -> fi.EndsWith ("FSharp.Core.dll"))
+      |> Seq.toList
+    assemblies |> shouldEqual [ scenarioTempPath "i001815-multiple-fsharp-core-dotnetcore-references" @@ "packages" @@ "Microsoft.FSharp.Core.netcore" @@ "lib" @@ "netstandard1.6" @@ "FSharp.Core.dll" ]
```
Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: I think this is wrong. I'm enumerating the list of resolved packages, what I'm missing here are that some packages are not pulled always. Is there an API to get this information easily (ie filter the list of packages first?)

File: integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/Project/project.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "net46": {
+      "dependencies": {
+        "FSharp.Core": "4.0.0.1",
+      },
+      "frameworkAssemblies": {
+        "mscorlib": "",
+        "System": "",
+        "System.Core": ""
+      }
+    },
```
Comment (matthid) [inline integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/Project/project.json]: @enricosada This way your package depends on both `FSharp.Core` and `Microsoft.FSharp.Core.netcore` ....

File: integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/OtherProject/testproject.csprojtemplate [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+    <When Condition="$(TargetFrameworkIdentifier) == '.NETStandard' And $(TargetFrameworkVersion) == 'v1.6'">
+      <ItemGroup>
+        <Reference Include="FSharp.Core">
+          <HintPath>..\packages\Microsoft.FSharp.Core.netcore\lib\netstandard1.6\FSharp.Core.dll</HintPath>
+          <Private>True</Private>
+          <Paket>True</Paket>
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
```
Comment (matthid) [inline integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/OtherProject/testproject.csprojtemplate]: As we can see here Paket actually is doing the correct thing ... Now I need to know how to get this InstallModel magic via the public Paket API :)

File: integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/Project/project.json (hunk 2 above)
Comment (enricosada) [inline integrationtests/scenarios/i001815-multiple-fsharp-core-dotnetcore-references/before/Project/project.json]: The two importing netstandard are not needed, so no issue. But yes, understood, thx @matthid for example

File: integrationtests/Paket.IntegrationTests/InstallSpecs.fs [hunk 4]
```diff
@@ -541,7 +541,9 @@ let ``#1815 duplicate fsharp core reference when using netstandard1.6``() =
       |> Seq.toList
       |> Paket.LoadingScripts.PackageAndAssemblyResolution.getPackageOrderResolvedPackage
       |> Seq.collect (fun p ->
-        let installModel = paketDependencies.GetInstalledPackageModel(group, p.Name.ToString())
+        let installModel =
+          paketDependencies.GetInstalledPackageModel(group, p.Name.ToString())
+            .ApplyFrameworkRestrictions(Requirements.getRestrictionList p.Settings.FrameworkRestrictions)
```
Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: @forki Is this the correct thing to do, I have no idea. Should the public API already add the restrictions here?

Anyway thanks for bringing me on the right track!

Are there other things caller need to do?

File: integrationtests/Paket.IntegrationTests/InstallSpecs.fs (hunk 1 above)
Comment (forki) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: why are you using the API? What is the purpose of this?

Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: I try to find the references I need to provide for the FSharp.Compiler.Service (see my FAKE dotnetcore PR)

In any case paket API I think paket should provide the same info via API as well, don't you think?

Comment (forki) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: I looks like you don't query the installmodel for a specifc framework.

Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: Isn't that the framework variable (set to netstandard 1.6)?

Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: ah ok this is the working code (with the fix) github replaced the code and left the comment

Comment (matthid) [inline integrationtests/Paket.IntegrationTests/InstallSpecs.fs]: see https://github.com/fsprojects/Paket/commit/5e795de33af67c70e31bb8e2bb7238793f45f2a3#commitcomment-18294517

**Comment by forki** [comment]
your test is working on my machine - so what is wrong?

**Comment by forki** [comment]
I merged it since it's only a test and I changed couple of lines to make it work in ncrunch

**Pull request merged**